### PR TITLE
Add tests to stripe_test

### PR DESCRIPTION
**Pull request by dgkutnic**
This PR adds various tests to `stripe_test.cc`:
  * Softmax
  * Elementwise Ops
  * Layer Normalization
  * Batch Normalization

### CODE REVIEW

File: pmlc/dialect/stripe/stripe_test.cc [hunk 1]
```diff
@@ -156,6 +156,68 @@ TEST_P(TranscodeTest, MaxPool2d) {
...
+TEST_P(TranscodeTest, MatMul) {
+  using plaidml::edsl::LogicalShape;
+  LogicalShape A(PLAIDML_DATA_FLOAT32, {16, 16});
+  LogicalShape B(PLAIDML_DATA_FLOAT32, {16, 16});
+  using vertexai::tile::lib::LoadMatMul;
+  auto ri = LoadMatMul("matmul", A, B);
+  RunTest(ri, GetParam());
+}
+
+// These two tests fail with invalid tensor dimensions if the location parameter for RunTest is set to 0.
```
Comment (tzerrell) [inline pmlc/dialect/stripe/stripe_test.cc]: We'll need to consult with @flaub but there's a good chance that's a real bug

File: pmlc/dialect/stripe/stripe_test.cc [hunk 2]
```diff
@@ -156,6 +156,68 @@ TEST_P(TranscodeTest, MaxPool2d) {
...
+  using plaidml::edsl::LogicalShape;
+  LogicalShape A(PLAIDML_DATA_FLOAT32, {1, 64, 64, 32});
+  using vertexai::tile::lib::LoadLayerNorm4dAx2;
+  auto ri = LoadLayerNorm4dAx2("layer_norm", A);
+  RunTest(ri, 1);
+}
+
+TEST_P(TranscodeTest, BatchNormalization) {
+  using plaidml::edsl::LogicalShape;
+  LogicalShape A(PLAIDML_DATA_FLOAT32, {1, 64, 64, 32});
```
Comment (tzerrell) [inline pmlc/dialect/stripe/stripe_test.cc]: Batch normalization with batch size 1 is meaningless in a practical sense. The calculations this will demonstrate are probably perfectly reasonable though.

File: tile/lib/lib.cc [hunk 3]
```diff
@@ -493,6 +493,22 @@ lang::RunInfo LoadSoftmax(const std::string& name,      //
...
+  return ((I - M) * G / sqrt(V + E)) + B;
+}
+
+lang::RunInfo LoadBatchNormalization(const std::string& name,      //
+                                     const LogicalShape& input) {  //
+  auto I = Placeholder(input);
+  auto M = Placeholder(input);
+  auto V = Placeholder(input);
+  auto G = Placeholder(input);
+  auto B = Placeholder(input);
```
Comment (tzerrell) [inline tile/lib/lib.cc]: I would expect these to have the batch dimension changed to 1 rather than left the same as on the input. In the place you're using this, it already is size 1, but that is itself a bit weird. It would probably also be good to have broadcast tested and this would be a natural place to do it. So perhaps drop the first dimension from `input` to create the shape for these Tensors and see if things still work?

File: pmlc/dialect/stripe/stripe_test.cc (hunk 2 above)
Comment (dgkutnic) [reply pmlc/dialect/stripe/stripe_test.cc]: Fixed this -- batch size is now 16.

**Comment by flaub** [comment]
Well I'm OK with landing as is. But @dgkutnic if you want to also try to fix the bug, that's fine with me too.
